src/sourcepoint.ts

```diff
@@ -0,0 +1,171 @@
+					if (message_type != frameworkMessageType) return;
+
+					// Event fires when a message is about to display.
+					//  TODO rename
+					mark('cmp-tcfv2-ui-displayed');
+				},
+
+				onMessageReceiveData: (message_type, data) => {
+					// Event fires when a message is displayed to the user and sends data about the message and campaign to the callback.
+					// The data sent to the callback is in the following structure:
+					log('cmp', `onMessageReceiveData ${message_type}`);
+					if (message_type != frameworkMessageType) return;
+
+					log('cmp', 'onMessageReceiveData ', data);
+					void resolveWillShowPrivacyMessage(data.messageId !== 0);
+				},
+
+				onMessageChoiceSelect: (
+					message_type,
+					choice_id,
+					choiceTypeID,
+				) => {
+					log(
+						'cmp',
+						`onMessageChoiceSelect message_type: ${message_type}`,
+					);
+					console.log();
+					if (message_type != frameworkMessageType) return;
+
+					log('cmp', `onMessageChoiceSelect choice_id: ${choice_id}`);
+					log(
+						'cmp',
+						`onMessageChoiceSelect choice_type_id: ${choiceTypeID}`,
+					);
+					if (
+						// https://documentation.sourcepoint.com/web-implementation/web-implementation/multi-campaign-web-implementation/event-callbacks#choice-type-id-descriptions
+						choiceTypeID === 11 ||
+						choiceTypeID === 13 ||
+						choiceTypeID === 15
+					) {
+						setTimeout(invokeCallbacks, 0);
+					}
+				},
+				onPrivacyManagerAction: function (message_type, pmData) {
+					log(
+						'cmp',
+						`onPrivacyManagerAction message_type: ${message_type}`,
+					);
+					if (message_type != frameworkMessageType) return;
+
+					log('cmp', `onPrivacyManagerAction ${pmData}`);
+				},
+				onMessageChoiceError: function (message_type, err) {
+					log('cmp', `onMessageChoiceError ${message_type}`);
+					if (message_type != frameworkMessageType) return;
+
+					log('cmp', `onMessageChoiceError ${err}`);
+				},
+				onPMCancel: function (message_type) {
+					log('cmp', `onPMCancel ${message_type}`);
+					if (message_type != frameworkMessageType) return;
+				},
+				onSPPMObjectReady: function () {
+					log('cmp', 'onSPPMObjectReady');
+				},
+				onError: function (
+					message_type,
+					errorCode,
+					errorObject,
+					userReset,
+				) {
+					log('cmp', `errorCode: ${message_type}`);
+					if (message_type != frameworkMessageType) return;
+
+					log('cmp', `errorCode: ${errorCode}`);
+					log('cmp', errorObject);
+					log('cmp', `userReset: ${userReset}`);
+				},
+			},
+		},
+	};
+
+	if (framework === 'tcfv2') {
+		window._sp_.config.gdpr.targetingParams = {
+			framework,
+		};
+	} else {
+		window._sp_.config.ccpa.targetingParams = {
+			framework,
+		};
+	}
+
+	const spLib = document.createElement('script');
+	spLib.id = 'sourcepoint-lib';
+	// TODO put this file in place!
+	spLib.src = `${ENDPOINT}/unified/wrapperMessagingWithoutDetection.js`;
+
+	document.body.appendChild(spLib);
```

Comment:
I think [`loadScript`] from `@guardian/libs` could help, here?

[`loadScript`]: https://github.com/guardian/libs/blob/main/src/loadScript.README.md

Reply:
Good call!

Reply:
We decided not to do this for now because 
1. `loadScript` returns a promise which it would be natural for `init` to return as well. This would be a change to the cmp api which belongs in a separate PR and release.
2. `loadScript` expects there to be at least on script element in the document, breaking tests.